src/ui/components/NetworkMonitor/ResponseBody.tsx

```diff
@@ -0,0 +1,54 @@
+import { ResponseBodyData } from "@recordreplay/protocol";
+import React, { useState } from "react";
+import ReactJson from "react-json-view";
+import { TriangleToggle } from "./RequestDetails";
+import { RequestSummary } from "./utils";
+
+const FIVE_MEGABYTES = 5e6;
+
+const safeJSONParseBase64 = (maybeJSON: string) => {
+  try {
+    return [true, JSON.parse(atob(maybeJSON))];
+  } catch (e) {
+    console.log(maybeJSON);
+    console.log(e);
+    return [false, maybeJSON];
+  }
+};
+
+const ResponseBody = ({
+  request,
+  responseBodyParts,
+}: {
+  request?: RequestSummary;
+  responseBodyParts: ResponseBodyData[];
+}) => {
+  if (!request || !responseBodyParts?.length) {
+    return null;
+  }
+
+  const contentLength = request.requestHeaders.find(h => h.name.toLowerCase() === "content-length");
+  const [validJSON, content] = safeJSONParseBase64(responseBodyParts.map(p => p.value).join(""));
```

Comment:
There's absolutely no guarantee that these parts will be in order so this is a bit dangerous. You also can't concatenate base64 values together like this. You'll need to `atob` each one separately. See my next comment for more info.

Reply:
This should be all taken care of now!